components/places/src/bookmark_sync/store.rs

```diff
@@ -146,6 +144,7 @@ impl<'a> BookmarksStore<'a> {
                             d.merged_node.merge_state.upload_reason() != UploadReason::None)
                     })
                 ), &params)?;
+                tx.maybe_commit()?;
```

Comment:
It's late (and a Sunday! 😅) here, so we can chat more tomorrow, but...I'm curious what these `maybe_commit`s do? `update_local_items` (and `stage_local_items_to_upload`) can't be broken up into multiple transactions, or run separately. Everything they do can run as part of a larger transaction, but not in smaller chunks.

Otherwise, we'll corrupt the local tree if the main connection does any writes—since both the in-memory merged tree, and the `mergedTree` table, might no longer match what's in `moz_bookmarks`—or expose an inconsistent tree, if we commit after any of the triggers.

Reply:
yes, sorry about that - I noticed they were against temp tables, but didn't think it through wrt the triggers.